### PR TITLE
Schedules with API Pagination

### PR DESCRIPTION
**Pull request by Alex-Tideman**
## Description & motivation 💭 
Use API pagination for schedules so you can view more than 1,000 schedules. Since you cannot search schedules to narrow down what you are looking forward, this is necessary for users to find schedules if they have over 1,000.

### Screenshots (if applicable) 📸 
Screen Shot 2024-01-04 at 12 12 34 PM
Screen Shot 2024-01-04 at 12 26 53 PM

### Design Considerations 🎨 

- [x] Search only finds current page schedules

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### CODE REVIEW

File: src/lib/pages/schedules.svelte [hunk 1]
```diff
@@ -49,7 +36,7 @@
       {translate('common.schedules')}
     </h1>
   </div>
-  {#if hasSchedules && !createDisabled}
+  {#if !createDisabled}
```
Comment (laurakwhit) [inline src/lib/pages/schedules.svelte:39]: Removing the `hastSchedules` check here means that this button shows up in two places on the `No Schedules Found` page. Was there a reason it was removed? 

Screenshot 2024-01-03 at 1 03 19 PM

Comment (Alex-Tideman) [reply src/lib/pages/schedules.svelte:39]: We don't know the count of schedules anymore outside of the <ApiPagination> component (let:visibleItems are schedules, so only the children of ApiPagination know about the count). And that Create Schedule button is outside of it. So we could refactor a bunch of things to make it work, but it feels like it's not worth it and would add a lot of complexity/slots to ApiPagination.

Comment (Alex-Tideman) [reply src/lib/pages/schedules.svelte:39]: Or we can remove the button in the EmptyState

Comment (laurakwhit) [reply src/lib/pages/schedules.svelte:39]: Ah makes sense. I think that was added as part of a design request (to only show one on the page) but I can't find verification of that. We could also wrap that button in an additional `{#await fetchSchedules}` and check  `{#if schedules?.length}`.

File: src/lib/pages/schedules.svelte [hunk 2]
```diff
@@ -41,38 +41,38 @@
...
-      nextButtonLabel={translate('common.next')}
-    >
-      <svelte:fragment slot="action-top-left">
-        <div class="w-full xl:w-1/2">
+{#key namespace}
+  <ApiPagination
+    let:visibleItems
+    {onFetch}
+    {onError}
+    aria-label="schedules"
```
Comment (laurakwhit) [inline src/lib/pages/schedules.svelte:49]:
```suggestion
    aria-label={translate('common.schedules')}
```

File: src/lib/pages/schedules.svelte [hunk 3]
```diff
@@ -41,38 +41,38 @@
...
+    pageSizeSelectLabel={translate('common.per-page')}
+    nextButtonLabel={translate('common.next')}
+    previousButtonLabel={translate('common.previous')}
+    emptyStateMessage={translate('schedules.empty-state-title')}
+    fallbackErrorMessage={'Error fetching schedules'}
```
Comment (laurakwhit) [inline src/lib/pages/schedules.svelte:54]: Looks like we use "Error fetching schedules" a few places, maybe worth defining in i18n?

File: src/lib/pages/schedules.svelte [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
+      err?.body?.message ??
+      `Error fetching schedules: ${err.status}: ${err.statusText}`);
 
   let coreUser = coreUserStore();
   $: createDisabled = $coreUser.namespaceWriteDisabled(namespace);
 
+  $: onFetch = () => fetchPaginatedSchedules(namespace);
+
   let search = '';
   $: filteredSchedules = (schedules: ScheduleListEntry[]) =>
```
Comment (laurakwhit) [inline src/lib/pages/schedules.svelte:36]: Just want to clarify that this is expected behavior. When showing filtered schedules the number of items doesn't change (e.g. only two items are shown, but pagination still says 1-3) ⬇️ 
 
Screenshot 2024-01-04 at 12 53 05 PM

**Review by laurakwhit — APPROVED** [review]
One clarification and two small suggestions, but otherwise lgmt 🚀

File: src/lib/pages/schedules.svelte (hunk 2 above)
Comment (Alex-Tideman) [reply src/lib/pages/schedules.svelte:49]: 👍

File: src/lib/pages/schedules.svelte (hunk 3 above)
Comment (Alex-Tideman) [reply src/lib/pages/schedules.svelte:54]: 👍

File: src/lib/pages/schedules.svelte (hunk 4 above)
Comment (Alex-Tideman) [reply src/lib/pages/schedules.svelte:36]: We don't get a total from the api. 

For the total when filtering... totally agree it's not ideal but I think for now we'll live with it. There's a whole lot of edge cases if we update the pagination store with a filter.

Comment (laurakwhit) [reply src/lib/pages/schedules.svelte:36]: Yeah, I figured it would be a can of worms. I think it's fine for now 👍

**Pull request merged**